deployments/charts/kuma/templates/_helpers.tpl

```diff
@@ -173,6 +173,9 @@ returns: formatted image string
 {{- end -}}
 
 {{- define "kuma.defaultEnv" -}}
+{{ if (and (eq .Values.controlPlane.environment "universal") (not (eq .Values.controlPlane.mode "global"))) }}
+  {{ fail "You can only run universal mode on kubernetes in a global mode" }}
+{{ end }}
```

Comment:
I don't think this limitation is required. You could run a zone universal control-plane on k8s no?

Reply:
https://github.com/kumahq/kuma/pull/5913#discussion_r1095827859 - same answer as here, let's move the conversation there